notebook2/notebook/notebook.py

```diff
@@ -301,283 +175,210 @@ def pod_condition_for_ui(conds):
-@requires_auth()
-def auth(requested_pod_uuid):
-    notebook = session.get('notebook')
-
-    if notebook is not None and notebook['pod_uuid'] == requested_pod_uuid:
-        res = flask.make_response()
-        res.headers['pod_ip'] = f"{notebook['pod_ip']}:{POD_PORT}"
-        return res
-
-    return '', 404
-
-
-def get_all_workers():
-    return k8s.list_namespaced_pod(
-        namespace='default',
-        watch=False,
-        label_selector='app=notebook2-worker',
         _request_timeout=KUBERNETES_TIMEOUT_IN_SECONDS)
 
-
-@app.route('/workers')
-@requires_auth()
-def workers():
-    if not session.get('admin'):
-        return redirect(external_url_for('admin-login'))
-
-    return render_template('workers.html',
-                           workers=get_all_workers(),
-                           workers_url=external_url_for('workers'),
-                           leader_instance=INSTANCE_ID)
-
-
-@app.route('/workers/<pod_name>/delete')
-@requires_auth()
-def workers_delete(pod_name):
-    if not session.get('admin'):
-        return redirect(external_url_for('admin-login'))
-
-    delete_worker_pod(pod_name)
-
-    return redirect(external_url_for('workers'))
+    for pod in pods.items:
+        if pod.metadata.deletion_timestamp is None:
+            return pod_to_ui_dict(pod)
 
 
-@app.route('/workers/delete-all-workers', methods=['POST'])
-@requires_auth()
-def delete_all_workers():
-    if not session.get('admin'):
-        return redirect(external_url_for('admin-login'))
-
-    for pod_name in get_all_workers():
-        delete_worker_pod(pod_name)
-
-    return redirect(external_url_for('workers'))
-
-
-def delete_worker_pod(pod_name):
+async def delete_worker_pod(k8s, pod_name):
     try:
-        k8s.delete_namespaced_pod(
+        await k8s.delete_namespaced_pod(
             pod_name,
-            'default',
+            NOTEBOOK_NAMESPACE,
             _request_timeout=KUBERNETES_TIMEOUT_IN_SECONDS)
     except kube.client.rest.ApiException as e:
         log.info(f'pod {pod_name} already deleted {e}')
 
 
-@app.route('/admin-login', methods=['GET'])
-@requires_auth()
-def admin_login():
-    return render_template('admin-login.html',
-                           form_action_url=external_url_for('admin-login'))
-
-
-@app.route('/admin-login', methods=['POST'])
-@requires_auth()
-def admin_login_post():
-    if request.form['password'] != ADMIN_PASSWORD:
-        return '403 Forbidden', 403
+@routes.get('/healthcheck')
+async def healthcheck(request):  # pylint: disable=unused-argument
+    return web.Response()
 
-    session['admin'] = True
 
-    return redirect(external_url_for('workers'))
+def base_context(userdata):
+    return {
+        'base_path': deploy_config.base_path('notebook2'),
+        'auth_base_url': deploy_config.external_url('auth', ''),
+        'userdata': userdata
+    }
 
 
-@app.route('/worker-image')
-def worker_image():
-    return WORKER_IMAGE, 200
+@routes.get('')
+@routes.get('/')
```

Comment:
duplicate

Reply:
This is not duplicate.  It's a sort of workaround for aiohttp.  We're using subapps when running on in a dev namespace (to add the /ns/svc path prefix), and if you navigate to internal.hail.is/ns/svc, the path is '' in the subapp and it doesn't match '/'.

Reply:
oh weird